app/assets/javascripts/hyrax/collections.js

```diff
@@ -240,27 +240,29 @@ Blacklight.onLoad(function () {
     };
     var $deleteWordingTarget = $('#selected-collections-delete-modal .pluralized');
 
-    tableRows.each(function(i, row) {
-      checkbox = $(row).find('td:first input[type=checkbox]');
-      if (typeof checkbox[0] !== "undefined") {
-        if (checkbox[0].checked) {
-          numRowsSelected++;
-        }
-      }
-    });
+    var canDeleteAll = true;
+    var selectedInputs = $('#documents table.collections-list-table tbody tr')
+      // Get all inputs in the table
+      .find('td:first input[type=checkbox]')
+      // Filter to those that are checked
+      .filter((i, checkbox) => checkbox.checked)
+
+    var cannotDeleteInputs = selectedInputs.filter((i, checkbox) => (checkbox.dataset.hasaccess === "false"))
```

Comment:
Change `.filter((i, checkbox) => (checkbox.dataset.hasaccess === "false"))` to `.filter(function(i, checkbox) { return checkbox.dataset.hasaccess === "false"; })`